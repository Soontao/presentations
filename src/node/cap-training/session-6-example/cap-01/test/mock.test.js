
const cds = require("@sap/cds")

const path = require("path")

describe('Mock Test Suite', () => {

  const mockDb = cds.db = new cds.Service('db') // mock service

  const run = mockDb.run = jest.fn()

  beforeEach(() => {
    run.mockClear()
  })


  /**
   * 
   * @returns {Promise<import("@sap/cds/apis/services").Service>}
   */
  const connectToService = () => cds.connect.to("ImportantService", {
    impl: cds.ApplicationService,
    model: path.join(__dirname, "../srv/ImportantService.cds")
  });

  it('should support connect to ImportantService', async () => {
    const s = await connectToService()
    expect(s).toBeInstanceOf(cds.ApplicationService)

  });

  it('should support insert data', async () => {
    const s = await connectToService()
    const result = await s.run(INSERT.into("Table3s").entries({ ID: '651db316-edfa-11ec-8ea0-0242ac120002', c1: "v1" }))
    expect(result).toMatchSnapshot() // check snapshot at __snapshots__ folder
  });

  it('should support soft deletion', async () => {
    const s = await connectToService()
    await s.run(DELETE.from("Table3s", '651db316-edfa-11ec-8ea0-0242ac120002'))
    expect(run.mock.calls).toMatchSnapshot() // check snapshot at __snapshots__ folder
  });


});