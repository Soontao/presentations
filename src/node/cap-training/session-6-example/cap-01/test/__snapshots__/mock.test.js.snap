// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock Test Suite should support insert data 1`] = `
Object {
  "ID": "651db316-edfa-11ec-8ea0-0242ac120002",
  "c1": "v1",
}
`;

exports[`Mock Test Suite should support soft deletion 1`] = `
Array [
  Array [
    UPDATE {
      "UPDATE": Object {
        "data": Object {
          "deleted": true,
        },
        "entity": Object {
          "ref": Array [
            Object {
              "id": "ImportantService.Table3s",
              "where": Array [],
            },
          ],
        },
      },
    },
    Object {
      "deleted": true,
    },
  ],
]
`;
